esmvaltool/diag_scripts/aerosols/aero_utils.py

```diff
@@ -0,0 +1,270 @@
+                "Altitude" coordinate')
+        else:
+
+            # Store the min and max altitudes from cube data so that user
+            # cannot request points located below/ above that.
+            # Will extract =min/max if beyond limits.
+            hgt_min = icube.coord('altitude').points.min()
+            hgt_max = icube.coord('altitude').points.max()
+
+    # ---------- Finished checks -- begin processing -------------------------
+
+    # If level specified, extract slice first
+    if 'level' in keys:
+
+        if icube.coord('model_level_number'):
+            icube = icube.extract(
+                iris.Constraint(model_level_number=keys['level']))
+
+        else:
+
+            if icube.coord('pseudo_level'):
+
+                icube = icube.extract(
+                    iris.Constraint(pseudo_level=keys['level']))
+
+    # Extract values for specified points lat/lon
+
+    # ******** Does not seem to handle multiple points if 3-D *************
+
+    data_out = []
+
+    for n in np.arange(len(pt_lat1)):
+        pt_coords = [('latitude', pt_lat1[n]), ('longitude', pt_lon1[n])]
+
+        if 'nearest' in keys and keys['nearest']:
+            tcube = icube.interpolate(pt_coords, iris.analysis.Nearest())
+
+        else:
+            tcube = icube.interpolate(pt_coords, iris.analysis.Linear())
+
+        # Extract at requested height
+
+        if 'height' in keys:
+            pt = max(pt_hgt[n], hgt_min)
+            pt = min(pt_hgt[n], hgt_max)
+            pt_coords = [('altitude', pt)]
+
+            if 'nearest' in keys and keys['nearest']:
+                tcube = tcube.interpolate(pt_coords, iris.analysis.Nearest())
+
+            else:
+                tcube = tcube.interpolate(pt_coords, iris.analysis.Linear())
+
+        data_out.append(tcube.data)
+
+    return data_out
+
+
+def calc_mass(pstar, prho, ptheta_top, mmr, nlev):
+    """Calculates mass in model layer from mass mixing ratios.
+
+    Method
+    ------
+    Simply calculates mass in layer using mmr*rho*dz = mmr*g*(-dp)
+       - at lowest level uses dp = p2 - pstar
+       - at highest level uses dp = p_theta(top) - p_rho(top_rho_lev)
+    Orig/IDL: Stephanie Woodward, 2003
+
+    Parameters
+    ----------
+    pstar :
+    prho :
+    ptheta :
+    mmr : Iris cube
+       Iris cube containing mass mixing ratio data.
+    nlev : int (optional)
+
+    Returns
+    -------
+    tmp : Iris cube
+        Iris cube containing mass data.
+    """
+
+    tmp = mmr.copy()
+    tmp.data[:, :, :] = 0.
+    tmp[0, :, :] = mmr[0, :, :] * (pstar - prho[1, :, :]) / earth_g
+
+    # Loop over model levels. Range will iterate up to nlev-2
+    for i in range(1, nlev - 1):
+        tmp[i, :, :] = mmr[i, :, :] * (prho[i, :, :] -
+                                       prho[i + 1, :, :]) / earth_g
+
+    tmp[nlev - 1, :, :] = mmr[nlev - 1, :, :] * (prho[nlev - 1, :, :] -
+                                                 ptheta_top) / earth_g
+
+    return tmp
```

Comment:
```suggestion
```
I don't see this function used anywhere - can it be removed?

Reply:
Done